File: python/motor.py
from kendaraan import kendaraan

class motor(kendaraan):
    def __init__(self, id_kendaraan="", merk="", plat_nomor="", tahun_produksi=0, warna="", jumlah_roda=0, tipe=""):
        super().__init__(id_kendaraan, merk, plat_nomor, tahun_produksi, warna, jumlah_roda)
        self.tipe = tipe

    # Getter method
    def get_tipe(self):
        return self.tipe

    # Setter method
    def set_tipe(self, tipe):
        self.tipe = tipe

File: cpp/motor.cpp
#pragma once
#include <bits/stdc++.h>
#include <iostream>
#include <string>
#include "kendaraan.cpp"

using namespace std;

class motor : public kendaraan{
    private:
        string tipe;

    // CONSTRUCTOR
    public:
        motor(){
            this->tipe = "";
        }

        motor(string id_kendaraan, string merk, string plat_nomor, int tahun_produksi, string warna, int jumlah_roda, string tipe)
        : kendaraan(id_kendaraan, merk, plat_nomor, tahun_produksi, warna, jumlah_roda){
            this->tipe = tipe;
        }

        // DESTRUCTOR
        ~motor(){
        }

    // GETTER
    string get_tipe(){
        return tipe; 
    }


    // SETTER
    void set_tipe(string tipe){
        this->tipe = tipe;
    }

};
